src/Template/Template.php

```diff
@@ -305,6 +294,71 @@ public function section($name, $default = null)
         return $this->sections[$name];
     }
 
+    /**
+     * Allows templates that are fetching this template to
+     * fetch the sections as well.
+     *
+     * @return array
+     */
+    protected function getSections()
+    {
+        return $this->sections;
+    }
+
+    /**
+     * Joins another Template's sections into this taking in
+     * consideration the template's section mode.
+     *
+     * @param Template $template
+     * @return void
+     */
+    protected function joinSections($template)
+    {
+        foreach ($template->getSections() as $sectionName => $sectionContent) {
+            $this->pushSection($template->sectionMode, $sectionName, $sectionContent);
+        }
+    }
+
+    /**
+     * Pushes the given section content to the sections array.
+     *
+     * @param int $sectionMode
+     * @param string $sectionName
+     * @param string $sectionContent
+     *
+     * @return void
+     */
+    private function pushSection($sectionMode, $sectionName, $sectionContent)
```

Comment:
I think `addSection` would make for a better name:

```suggestion
    private function addSection($sectionMode, $sectionName, $sectionContent)
```